From 87e70756cd72d999042275d0c379a495b5d21e88 Mon Sep 17 00:00:00 2001
From: Nikolay Aleksandrov <nikolay@redhat.com>
Date: Mon, 1 Sep 2014 13:33:20 -0400
Subject: [crypto] qat: Fix return value check in adf_chr_drv_create()

Message-id: <1409578401-26943-29-git-send-email-nikolay@redhat.com>
Patchwork-id: 90105
O-Subject: [RHEL7.1 PATCH BZ 1091130 28/29] crypto: qat - Fix return value check in adf_chr_drv_create()
Bugzilla: 1091130
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Herbert Xu <herbert.xu@redhat.com>

From: Wei Yongjun <yongjun_wei@trendmicro.com.cn>

In case of error, the function device_create() returns ERR_PTR()
and never returns NULL. The NULL test in the return value check
should be replaced with IS_ERR().

Signed-off-by: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
(cherry picked from commit 65fafbe9bddf49d53a2bbae92498909e3c5f964b)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/crypto/qat/qat_common/adf_ctl_drv.c b/drivers/crypto/qat/qat_common/adf_ctl_drv.c
index d97069b..6f7816e 100644
--- a/drivers/crypto/qat/qat_common/adf_ctl_drv.c
+++ b/drivers/crypto/qat/qat_common/adf_ctl_drv.c
@@ -111,7 +111,7 @@ static int adf_chr_drv_create(void)
  drv_device = device_create(adt_ctl_drv.drv_class, NULL,
        MKDEV(adt_ctl_drv.major, 0),
        NULL, DEVICE_NAME);
- if (!drv_device) {
+ if (IS_ERR(drv_device)) {
   pr_err("QAT: failed to create device\n");
   goto err_cdev_del;
  }
-- 
1.7.1